### ISSUE TITLE
As a blog author, I want to display a post preview on the index page

### ISSUE THREAD

**Issue by gaylin** [gaylin/gaylin.github.io#18]
I like the way several blogs display a small snippet of text near the blog title on the index page. I like how the snippet serves as a summary and lets readers decide which post to examine first if they don't want to read them in chronological order. 

In an ideal world I would want the snippet to be automatically generated unless I specifically told it either what the exact snippet should be or where the snippet should cut off if it merely started from the beginning of the post and went a certain length before the snippet cutoff.

**Comment by gaylin** [comment]
Okay, here's my logic...

If I wanted the automagically generate stuff, I think all I would need to add is the built in [`{{ post.excerpt }}` markup](http://jekyllrb.com/docs/posts/#post-excerpts) in my blog's `index.html` file wherever the formatting worked the best (maybe before the `</li>`). This would let jekyll use the first paragraph of the post as the excerpt.

But what if that automagic excerpt is just too long or I don't like the first paragraph as an intro and want more control? I think I could instead simply specify the separator by adding something like `excerpt_separator: "<!-- more -->"` to my `_config.yml` file to exactly control the break point. The system would look for `<!--more-->` in any post and take what appeared before it as the summary. Right?

But then the more I think about it, what if I can't decide what I'll want? What if I want to be able to do both? Or what if I forgot the separator entirely? Can I set it up to be smarter and help me out if I forget to specify an excerpt? I think so if I do this instead:

```
{% if post.content contains '<!--more-->' %}
    {{ post.content | split:'<!--more-->' | first }}
{% else %}
    {{ post.excerpt | strip_html }}
{% endif %}
```
If I'm understanding the way this works, then the `if` statement tells the system to check if I included the `<!--more-->` HTML comment tag in the actual blog post. The `split` and `first` bits in that mess tell the system to use the comment as a truncation point, taking what's before it as the summary. If the comment doesn't appear in the blog post, then the `else` statement kicks in. It gives me the option of just letting Jekyll do its job and just take the first paragraph of text. That little `strip_html` bit appears so I can get around formatting weirdness if the cutoff point happens in the middle of some markup (say there's example code or an overlong quote at the beginning of a post).

If I want to do the whole thing manually, overriding Jekyll's automagically generated excerpt, I think Jekyll's precedence rules mean that I can I do it simply by adding the excerpt by hand in the frontmatter of the blog post:

```
---
layout: post
title: "<blog post title>"
date: <YYYY-MM-DD>
excerpt: <excerpt text>
---
```

Then no matter what I have in the `index.html` file, having the excerpt in the front matter takes precedence.

I'm going to mull over before implementing anything. Gods I need a coding buddy to tell me where my logic is flawed.

**Comment by gaylin** [comment]
Tested the simple {{ post.excerpt }} addition and it works like a charm, except....formatting is ginormous.

**Comment by gaylin** [comment]
So the new wordpress-style separator worked great but the logic of precedence didn't. I'll have to noodle that out a bit more but for now I'm going to go ahead and insert the separator at will.

**Comment by gaylin** [comment]
and...it doesn't work. back to the drawing board on the logic.

**Comment by gaylin** [comment]
Closed via #25 despite my fooling around (and failing) with making it fancy in later PRs.
